### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by ChristianSchwarz**
This PR fixes some compile errors and removes dead code:
 - compile error: duplicate org.mockito.package-info.java in _src/main/java_ and _src/test/java_
 - ~~compile error: char enconding error in MatchersTest -> relates to #765~~
 - dead code: Constructors -> relates to #904

### CODE REVIEW

**Comment by TimvdLippe** [comment]
It seems to be compiling just fine on CI :O

**Comment by ChristianSchwarz** [comment]
Yes, the JDT compiler seems to be more strict...

**Comment by TimvdLippe** [comment]
The unicode issue should be fixed with the option encoding in the compilation step. Don't think that is necessary to be fixed here.

File: src/test/java/org/mockitousage/matchers/MatchersTest.java [hunk 1]
```diff
@@ -608,7 +608,7 @@ public void same_matcher_and_nulls() {
     public void nullable_matcher() throws Exception {
         // imagine a Stream.of(...).map(c -> mock.oneArg(c))...
         mock.oneArg((Character) null);
-        mock.oneArg(Character.valueOf('€'));
```
Comment (bric3) [inline src/test/java/org/mockitousage/matchers/MatchersTest.java]: That statement should work, there's no need to modify the source here. If something is wrong maybe it's the file encoding during the git clone or other editor that changed the encoding. On my clone : 

```
$ file src/test/java/org/mockitousage/matchers/MatchersTest.java
src/test/java/org/mockitousage/matchers/MatchersTest.java: UTF-8 Unicode c program text
```

**Comment by ChristianSchwarz** [comment]
Okay, I reverted the UTF-8 change.

**Pull request merged**